ping/ping.md

```diff
@@ -0,0 +1,62 @@
+# Ping <!-- omit in toc -->
+
+| Lifecycle Stage | Maturity                 | Status | Latest Revision |
+| --------------- | ------------------------ | ------ | --------------- |
+| 3A              | Recommendation | Active | r0, 2022-11-04  |
+
+Authors: [@marcopolo]
+
+Interest Group: [@marcopolo], [@mxinden], [@marten-seemann]
+
+[@marcopolo]: https://github.com/mxinden
+[@mxinden]: https://github.com/mxinden
+[@marten-seemann]: https://github.com/marten-seemann
+
+# Table of Contents <!-- omit in toc -->
+- [Protocol](#protocol)
+- [Diagram](#diagram)
+
+# Protocol
+
+The ping protocol is a simple request response protocol. The client opens a
+stream, sends a payload of 32 random bytes, and the server responds with the
+same 32 bytes on that stream. The client then measures the RTT from the time it
+wrote the bytes to the time it received the bytes. The client MAY repeat the
+process by sending another payload with random bytes on the same stream, so the
+server SHOULD loop and echo the next payload. The client SHOULD close the write
+side of the stream after sending the last payload, and the server SHOULD finish
```

Comment:
Why is it "SHOULD" rather than "MUST"?

If the server is allowed to not respond to pings (which is implied by the usage of "SHOULD"), then the protocol is pretty useless.

Of course, the client should be aware of the fact that the server can in practice do something else than this text says, but this is off-topic for a specification. The word "MUST" implicitly means "MUST ... in order to comply to this specification". The fact that you might be connected to a peer that doesn't conform to the specification is obvious and implied, and doesn't need to be mentioned in the specification.